Assets/Mirror/Components/NetworkAnimator.cs

```diff
@@ -174,54 +174,62 @@ internal void HandleAnimTriggerMsg(int hash)
             animator.SetTrigger(hash);
         }
 
-        bool WriteParameters(NetworkWriter writer, bool autoSend)
+        uint GetDirtyBits()
         {
             uint dirtyBits = 0;
-            // Save the position in the writer where to insert the dirty bits
-            int dirtyBitsPosition = writer.Position;
-            // Reserve the space for the bits
-            writer.Write(dirtyBits);
             for (int i = 0; i < parameters.Length; i++)
             {
                 AnimatorControllerParameter par = parameters[i];
+                bool changed = false;
                 if (par.type == AnimatorControllerParameterType.Int)
                 {
                     int newIntValue = animator.GetInteger(par.nameHash);
-                    if (newIntValue != lastIntParameters[i])
-                    {
-                        writer.WritePackedUInt32((uint) newIntValue);
-                        dirtyBits |= 1u << i;
-                        lastIntParameters[i] = newIntValue;
-                    }
+                    changed = newIntValue != lastIntParameters[i];
+                    if (changed) lastIntParameters[i] = newIntValue;
                 }
                 else if (par.type == AnimatorControllerParameterType.Float)
                 {
                     float newFloatValue = animator.GetFloat(par.nameHash);
-                    if (Mathf.Abs(newFloatValue - lastFloatParameters[i]) > 0.001f)
-                    {
-                        writer.Write(newFloatValue);
-                        dirtyBits |= 1u << i;
-                        lastFloatParameters[i] = newFloatValue;
-                    }
+                    changed = Mathf.Abs(newFloatValue - lastFloatParameters[i]) > 0.001f;
+                    if (changed) lastFloatParameters[i] = newFloatValue;
```

Comment:
I know we have been using ifs on the same line in a few places.   I am not a big fan of that because I cannot put a breakpoint in the statement.

@vis2k do you have an opinion on this?

Reply:
if (thing)
{
}
is pretty much used everywhere in mirror, except for a few early returns

Reply:
I'll change it to be that way.

Reply:
Not a fan of { }  either,  they are redundant,  the indentation already tells you it is inside the ifs

Reply:
One of my next PRs will involve a change here anyways, but there it is.